app/views/ChangeAvatarView/index.tsx

```diff
@@ -0,0 +1,196 @@
+import StatusBar from '../../containers/StatusBar';
+import { useTheme } from '../../theme';
+import SafeAreaView from '../../containers/SafeAreaView';
+import * as List from '../../containers/List';
+import styles from './styles';
+import { useAppSelector } from '../../lib/hooks';
+import { getUserSelector } from '../../selectors/login';
+import Avatar from '../../containers/Avatar';
+import AvatarUrl from './AvatarUrl';
+import Button from '../../containers/Button';
+import I18n from '../../i18n';
+import { ChatsStackParamList } from '../../stacks/types';
+import { IAvatar } from '../../definitions';
+import { Services } from '../../lib/services';
+import AvatarSuggestion from './AvatarSuggestion';
+import log from '../../lib/methods/helpers/log';
+
+const ChangeAvatarView = () => {
+	const [avatar, setAvatarState] = useState<IAvatar>();
+
+	const [textAvatar, setTextAvatar] = useState('');
+	const [saving, setSaving] = useState(false);
+	const { colors } = useTheme();
+	const { user, serverVersion } = useAppSelector(state => ({
+		user: getUserSelector(state),
+		isMasterDetail: state.app.isMasterDetail,
+		serverVersion: state.server.version
+	}));
+
+	const avatarUrl = useRef<string | undefined>('');
+
+	const navigation = useNavigation<StackNavigationProp<ChatsStackParamList, 'ChangeAvatarView'>>();
+	const { fromUser, titleHeader, room, t } = useRoute<RouteProp<ChatsStackParamList, 'ChangeAvatarView'>>().params;
+
+	useLayoutEffect(() => {
+		navigation.setOptions({
+			title: titleHeader || I18n.t('Avatar')
+		});
+	}, [titleHeader, navigation]);
+
+	useEffect(() => {
+		navigation.addListener('beforeRemove', e => {
+			if (!avatarUrl.current) {
+				return;
+			}
+
+			e.preventDefault();
+
+			showConfirmationAlert({
+				title: I18n.t('Discard_changes'),
+				message: I18n.t('Discard_changes_description'),
+				confirmationText: I18n.t('Discard'),
+				onPress: () => {
+					navigation.dispatch(e.data.action);
+				}
+			});
+		});
+	}, [navigation]);
+
+	const setAvatar = (value?: IAvatar) => {
+		avatarUrl.current = value?.url;
+		setAvatarState(value);
+	};
+
+	const submit = async () => {
+		try {
+			setSaving(true);
+			if (!fromUser && room?.rid) {
+				// Change Rooms Avatar
+				await Services.saveRoomSettings(room.rid, { roomAvatar: avatar?.data });
+			} else if (avatar?.url) {
+				// Change User's Avatar
+				await Services.setAvatarFromService(avatar);
+			} else if (textAvatar) {
+				// Change User's Avatar
+				await Services.resetAvatar(user.id);
+			}
+			setSaving(false);
+			avatarUrl.current = '';
+			return navigation.goBack();
+		} catch (e) {
+			log(e);
+			setSaving(false);
+			if (!fromUser && room?.rid) {
+				return handleError(e, 'saveRoomSettings', 'changing_avatar');
+			}
+			if (textAvatar) {
+				return handleError(e, 'resetAvatar', 'changing_avatar');
+			}
+			return handleError(e, 'setAvatarFromService', 'changing_avatar');
+		}
+	};
+
+	const resetAvatar = () => {
+		setAvatar(undefined);
```

Comment:
Bad practice.
You're mixing null and undefined usage.
Stick to null here by initializing your useState with null.